### ISSUE TITLE
Meter element states

### ISSUE THREAD

**Issue by lukewarlow** [openui/open-ui#1130]
I raised an [issue](https://github.com/w3c/csswg-drafts/issues/11336) in the csswg but it feels like it'd be good to discuss here too.
 
To be able to style a meter element we need some way to style based on how optimal the value of the meter element is.

WebKit (and blink) uses different pseudo elements and their naming is `::-webkit-meter-even-less-good-value`, `::-webkit-meter-suboptimum-value` and `::-webkit-meter-optimum-value`.

Firefox has a single `::-moz-meter-bar` pseudo element and instead applies pseudo classes to `<meter>` these are `:-moz-meter-sub-sub-optimum`, `:-moz-meter-sub-optimum` and `:-moz-meter-optimum`.

It would be good for us to discuss which of these approaches is more appropriate and to bikeshed names.

There's also an [interop issue](https://github.com/whatwg/html/issues/3520) between WebKit/Blink and Gecko on the algorithm for determining these states. Which we should fix too.

**Comment by lukewarlow** [comment]
As these are different states rather than inherently different structured elements pseudo classes seems the correct approach (though it's possible I'm missing a use case that WebKit allows that Firefox doesn't).

I suspect we can drop the meter prefix and just go with something like `:sub-sub-optimum`, `:sub-optimum`, `:optimum`? But naming would need bikeshedding as neither Firefox's nor WebKit's seem great.

**Comment by mfreed7** [comment]
I vote for `meter-value-is-less-good-than-the-not-quite-perfect-value-at-least-for-now`

**Comment by mfreed7** [comment]
Reminder: the still-open agenda item for this issue is to brainstorm better names for the pseudo classes. Note [the rules for meter coloring](https://html.spec.whatwg.org/multipage/form-elements.html#the-meter-element:~:text=UA%20requirements%20for%20regions%20of%20the%20gauge) at least according to the spec. Per that text, there are three different regions you can be in: the "optimum" region, the "sub-optimum" region, and the "even less good" region (yes this is the spec wording). The way you get "even less good" is when the `optimum` attribute is lower than the `low` or higher than the `high`, then the region at "the other end" is considered "even less good".

With that in mind, here are the pseudo classes I've heard mentioned:

- :sub-optimum
- :suboptimum-value
- :even-less-good-value
- :optimum-value.
- :sub-sub-optimum
- :sub-optimum
- :optimum

What else can people think of?

**Comment by lukewarlow** [comment]
`:worst`, `:medium`, `:optimum` or something along those lines maybe? I wonder if any design systems have the meter concept and if they do have then what names do they use?

**Comment by sorvell** [comment]
`:optimal-range`, `:less-optimal-range`, `:least-optimal-range`

The word "range" makes clear the styling applies to a region and not just a specific point. The term "optimal" is used instead of "optimum" because optimal can be an adjective but optimum is not. If matching the attribute is preferred, this could just be `optimum-*`.

**Comment by lukewarlow** [comment]
Yeah the linguistics part of me very much prefers optimal rather than optimum so if we don't need to match the attribute name that'd be my preference.
